### PR TITLE
[#IOPAE-709] Impements Institution APIs

### PR DESCRIPTION
**Pull request by giamma1295**
Implements Institutions APIs
#### List of Changes
- Implements `GET api/institutions` API
- Implements `GET api/institutions/:institutionId` API
- Add Selfcare Institutions related `io-ts` types
- Add Selfcare institution rest client

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by giuseppedipinto — APPROVED** [review]

File: infra/src/backoffice_webapp.tf [hunk 1]
```diff
@@ -13,6 +13,8 @@ locals {
     # NextAuthJS
     NEXTAUTH_URL    = "https://selfcare.io.pagopa.it/" # FIXME: move into terraform.tfvars or use a terraform resource/data ref
     NEXTAUTH_SECRET = azurerm_key_vault_secret.bo_auth_session_secret.value
+
+    SELFCARE_API_KEY = data.azurerm_key_vault_secret.selfcare_api_key.value
```
Comment (rodoviti) [inline infra/src/backoffice_webapp.tf:17]: **chore:** add `var.backoffice_app.selfcare_base_path` tfvar and add its value as `SELFCARE_BASE_PATH` environment variable

File: apps/backoffice/src/lib/be/institutions/business.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  dataProtectionOfficer: institutionResource.dpoData,
+  //geographicTaxonomies: institutionResource.geographicTaxonomies,
+  rea: institutionResource.companyInformations?.rea,
+  shareCapital: institutionResource.companyInformations?.shareCapital,
+  businessRegisterPlace:
+    institutionResource.companyInformations?.businessRegisterPlace,
+  supportEmail: institutionResource.assistanceContacts?.supportEmail,
+  supportPhone: institutionResource.assistanceContacts?.supportPhone,
+  //imported: institutionResource.imported,
+  //logo: institutionResource.logo,
```
Comment (rodoviti) [inline apps/backoffice/src/lib/be/institutions/business.ts:52]: **question:** are these comments left here intentionally?

File: infra/src/backoffice_webapp.tf (hunk 1 above)
Comment (giamma1295) [reply infra/src/backoffice_webapp.tf]: 15eeeac7a7d56100cf951f344d1ca92aaad3bd6e

File: apps/backoffice/src/lib/be/institutions/business.ts (hunk 2 above)
Comment (giamma1295) [reply apps/backoffice/src/lib/be/institutions/business.ts]: Removed in cleanUp dbecc499cf52f87fed0bebcf542f935a88a96c5e

File: apps/backoffice/src/lib/be/selfcare-client.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import { BooleanFromString } from "@pagopa/ts-commons/lib/booleans";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import axios, { AxiosError } from "axios";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { flow, identity, pipe } from "fp-ts/lib/function";
+import * as t from "io-ts";
+//import { cache } from "react";
```
[the remark this answers is not in the document]
Comment (giamma1295) [reply apps/backoffice/src/lib/be/selfcare-client.ts]: Removed in cleanUp dbecc499cf52f87fed0bebcf542f935a88a96c5e

**Comment by gunzip** [comment]
consider splitting this kind of PRs into multiple ones (shorter) in the future

**Pull request merged**